django_rq/views.py

```diff
@@ -218,6 +219,29 @@ def job_detail(request, queue_index, job_id):
         job = Job.fetch(job_id, connection=queue.connection)
     except NoSuchJobError:
         raise Http404("Couldn't find job with this ID: %s" % job_id)
+    # job should have dependency but it got deleted we need to cleanup
+    try:
+        job.dependency
+    except Exception as e:
+        # remove the dependency association
+        job.connection.srem(job.dependents_key_for(dependency_id), job.id)
```

Comment:
The way this tries to use Redis commands to directly cleanup dependent jobs is making me nervous. I think we should first add `job.delete_dependents()` to RQ so it can be called here.